assignment4/count_islands.py

```diff
@@ -0,0 +1,94 @@
+class CountIslands():
+    def __init__(self):
+        self._clear_data()
+
+    def _clear_data(self):
+        """*_row_islands is a list of sets containing tile numbers that belong to some island in the previous row
+        """
+        self.prev_row_islands = []
+        self.new_row_islands = []
+        self.counter = 0
+        self._clear_iterators()
+
+    def _clear_iterators(self):
+        self.prev_row_i = 0
+        self.new_row_i = 0
+
+    def _create_island(self, tile_num):
+        """We either create new set (new island) or extend an existing one.
+        :param tile_num:
+        :return:
+        """
+        added = False
+        if not self.new_row_islands:
+            self.new_row_islands.append({tile_num})
+            return
+
+        for island in self.new_row_islands:
+            if tile_num - 1 in island:
+                island |= {tile_num}
```

Comment:
Yeah, this it not that clear, though I wanted to try out the set operations style :D